src/posit/connect/content.py

```diff
@@ -27,6 +29,13 @@ def __getitem__(self, key: Any) -> Any:
             return ContentItemOwner(params=self.params, **v)
         return v
 
+    @property
+    def oauth_associations(self) -> ContentItemAssociations:
+        if self.guid is None:
+            raise ValueError("ContentItem must have a guid to have associations")
```

Comment:
You should do this check as part of `ContentItemAssociation.__init__`.

Also, `self["guid"]` instead of `self.guid`. See https://github.com/posit-dev/posit-sdk-py/pull/271

Reply:
doing the check inside of the `__init__` does produce slightly rough behavior, i.e.

```

    @property
    def associations(self) -> IntegrationAssociations:
        return IntegrationAssociations(self.params, integration_guid=self["guid"])


```

if `self["guid"]` is unset, we will get a `KeyError` before we get into the constructor.

Reply:
Ah, I see. Disregard that.